lib/types.nix

```
@@ -275,6 +275,22 @@ rec {
       merge = mergeEqualOption;
     };
 
+    anyBool = mkOptionType {
+      name = "anyBool";
+      description = "boolean (any)";
+      descriptionClass = "noun";
+      check = isBool;
+      merge = loc: defs:
+        foldl'
+          (result: def:
+            # We don't need to worry about ||'s short-circuit behavior influencing strictness,
+            # because def.value is always evaluated before anyways
```

```suggestion
            # Under the assumption that .check always runs before merge, we can assume that all defs.*.value
            # have been forced, and therefore we assume we don't introduce order-dependent strictness here
```

Yes we need to worry. Don't take any of this for granted, e.g. we might make types first class.

---

I'll add a test for this (actually already wrote one with the idea that I'd have to add a `seq` to make it work. Realised it wasn't necessary when it ran successfully)